R/sass_cache.R

```diff
@@ -1,101 +1,143 @@
+#'
+#' After `sass_cache_set()` is called for a directory, any future calls to
+#' `sass_cache_get()` with that directory will return that specific cache
+#' object. This can be useful if you customize parameters for the cache object,
+#' like maximum size or age.
+#'
+#' @param dir A directory. If `NULL`, this function will return `NULL`.
+#' @param cache A [sass_file_cache()] object, or `NULL` if you don't want to use
+#'   a cache.
+#'
+#' @seealso [sass_file_cache()], [sass()]
+#'
+#' @keywords internal
 #' @export
-sass_cache_get <- function() {
-  if (!exists("cache", .globals, inherits = FALSE)) {
-    sass_cache_set(sass_file_cache())
+sass_cache_get <- function(dir) {
+  if (is.null(dir)) {
+    return(NULL)
   }
-  .globals$cache
+
+  dir <- fs::path_abs(dir)
+  if (!exists(dir, .caches, inherits = FALSE)) {
+    sass_cache_set(dir, sass_file_cache(dir))
+  }
+  .caches[[dir]]
 }
 
 #' @rdname sass_cache_get
 #' @export
-sass_cache_set <- function(cache = sass_file_cache()) {
+sass_cache_set <- function(dir, cache) {
   if (!is.null(cache) && !inherits(cache, "FileCache")) {
     stop("`cache` must be a FileCache object or NULL.")
   }
-  .globals$cache <- cache
+  .caches[[dir]] <- cache
+}
+
+#' Get default sass cache object
+#'
+#' @description
+#'
+#' Get the default sass cache object or directory, for the given context. The
+#' context depends on factors described below.
+#'
+#' `sass_default_cache()` first checks the `sass.cache` option. If it is set to
+#' `FALSE`, then this function returns `NULL`. If it has been set to a string,
+#' it is treated as a directory name, and this function returns a
+#' `sass_file_cache()` object using that directory. If the option has been set
+#' to a `sass_file_cache()` object, then it will return that object.
+#'
+#' In most cases, this function uses the user's cache directory, by calling
+#' `rappdirs::user_cache_dir("R-sass")`.
+#'
+#' If this function is called from a Shiny application, it will also look for a
+#' subdirectory named `cache/`. If it exists, it will use a directory named
+#' `cache/sass/` to store the cache.
+#'
+#' When running a Shiny application in a user R session, it will not create the
+#' `cache/` subdirectory, but it will use it if present. This scopes the cache
+#' to the application.
+#'
+#' With Shiny applications hosted on Shiny Server and Connect, it _will_ create
+#' a `cache/sass/` subdirectory, so that the cache is scoped to the application
+#' and will not interfere with another application's cache.
+#'
+#' @seealso [sass_cache_get()], [sass()]
+#'
+#' @export
+sass_default_cache <- function() {
+  cache_option <- getOption("sass.cache", default = TRUE)
+  if (is.null(cache_option) || identical(cache_option, FALSE)) {
+    return(NULL)
+
+  } else if (inherits(cache_option, "FileCache")) {
+    return(cache_option)
+
+  } else if (is.character(cache_option)) {
+    sass_cache_get(cache_option)
+  }
```

Comment:
```suggestion
  if (is.null(cache_option) || identical(cache_option, FALSE)) {
    return(NULL)
  }
  
  if (inherits(cache_option, "FileCache")) {
    return(cache_option)
  }
  
  if (is.character(cache_option)) {
    return(sass_cache_get(cache_option))
  }
```

Reply:
Any particular reason you prefer the if-if style to else-if? I don't feel strongly about it either way.

Reply:
I prefer it since it's more minimal and makes the intention (early returns) clearer.

This also fixes the `is.character()` case (looks like you wanted a `return()` there, right?)